File: event-sourcing/python/tests/event_builder.py
from __future__ import annotations

from datetime import datetime, timezone

from event_sourcing import (
    AccountOpened,
    MoneyDeposited,
    MoneyWithdrawn,
    AccountClosed,
    Money,
)

DEFAULT_ACCOUNT_ID = "ACC-001"
DEFAULT_OWNER_NAME = "Alice"
DEFAULT_TIMESTAMP = datetime(2026, 1, 1, tzinfo=timezone.utc)


def an_account_opened(
    account_id: str = DEFAULT_ACCOUNT_ID,
    owner_name: str = DEFAULT_OWNER_NAME,
    timestamp: datetime = DEFAULT_TIMESTAMP,
) -> AccountOpened:
    return AccountOpened(
        account_id=account_id, timestamp=timestamp, owner_name=owner_name
    )


def a_money_deposited(
    amount: float | int,
    account_id: str = DEFAULT_ACCOUNT_ID,
    timestamp: datetime = DEFAULT_TIMESTAMP,
) -> MoneyDeposited:
    return MoneyDeposited(
        account_id=account_id, timestamp=timestamp, amount=Money(amount)
    )


def a_money_withdrawn(
    amount: float | int,
    account_id: str = DEFAULT_ACCOUNT_ID,
    timestamp: datetime = DEFAULT_TIMESTAMP,
) -> MoneyWithdrawn:
    return MoneyWithdrawn(
        account_id=account_id, timestamp=timestamp, amount=Money(amount)
    )


def an_account_closed(
    account_id: str = DEFAULT_ACCOUNT_ID,
    timestamp: datetime = DEFAULT_TIMESTAMP,
) -> AccountClosed:
    return AccountClosed(account_id=account_id, timestamp=timestamp)

File: event-sourcing/typescript/tests/EventBuilder.ts
import { Money } from '../src/Money.js';
import {
  accountOpened,
  moneyDeposited,
  moneyWithdrawn,
  accountClosed,
  type AccountOpened,
  type MoneyDeposited,
  type MoneyWithdrawn,
  type AccountClosed,
} from '../src/Events.js';

const DefaultAccountId = 'ACC-001';
const DefaultOwnerName = 'Alice';
const DefaultTimestamp = new Date(Date.UTC(2026, 0, 1));

export function anAccountOpened(opts?: {
  accountId?: string;
  ownerName?: string;
  timestamp?: Date;
}): AccountOpened {
  return accountOpened(
    opts?.accountId ?? DefaultAccountId,
    opts?.ownerName ?? DefaultOwnerName,
    opts?.timestamp ?? DefaultTimestamp,
  );
}

export function aMoneyDeposited(
  amount: number,
  opts?: { accountId?: string; timestamp?: Date },
): MoneyDeposited {
  return moneyDeposited(
    opts?.accountId ?? DefaultAccountId,
    new Money(amount),
    opts?.timestamp ?? DefaultTimestamp,
  );
}

export function aMoneyWithdrawn(
  amount: number,
  opts?: { accountId?: string; timestamp?: Date },
): MoneyWithdrawn {
  return moneyWithdrawn(
    opts?.accountId ?? DefaultAccountId,
    new Money(amount),
    opts?.timestamp ?? DefaultTimestamp,
  );
}

export function anAccountClosed(opts?: {
  accountId?: string;
  timestamp?: Date;
}): AccountClosed {
  return accountClosed(
    opts?.accountId ?? DefaultAccountId,
    opts?.timestamp ?? DefaultTimestamp,
  );
}
